drivers/gles3/rasterizer_canvas_gles3.cpp

```diff
@@ -1683,6 +1682,10 @@ void RasterizerCanvasGLES3::canvas_render_items(Item *p_item_list, int p_z, cons
 						} else {
 							state.canvas_shader.set_uniform(CanvasShaderGLES3::SCREEN_PIXEL_SIZE, Vector2(1.0, 1.0));
 						}
+						if (state.using_skeleton) {
+							state.canvas_shader.set_uniform(CanvasShaderGLES3::SKELETON_TRANSFORM, state.skeleton_transform);
```

Comment:
Is this necessary if https://github.com/godotengine/godot/pull/34570 is merged? It looks to me like we would end up setting the same uniform twice.

Reply:
I was thinking the same, but I believe this is necessary here because these uniforms go to the lighting pass shader. On #34570 they go to the "regular" shader.

Reply:
Ah yes, good point.

Reply:
The uniforms are set regardless of whether it is a light pass or not here https://github.com/godotengine/godot/blob/master/drivers/gles3/rasterizer_canvas_gles3.cpp#L1606

The canvas_shader is not rebound during the light passes so the uniform should still be up to date for each light.

Reply:
@MadEqua ping. The extra setting of the uniform is unnecessary here, the shader is the same as the uniforms are set immediately above, whether in a light pass or not.

Reply:
Sorry, this went under my radar.

I might be misunderstanding something but the shader with lighting active is being bound after that, here:

https://github.com/godotengine/godot/blob/318c69351624f7794c51b5385d252af397c0404a/drivers/gles3/rasterizer_canvas_gles3.cpp#L1677-L1679

and it needs the skeleton transforms to work correctly. That's why the issue still reproduces after merging #34570.

Reply:
Ah, I missed the ``bind()`` call. Thanks for bearing with me here. Looks good then.